ygot/struct_validation_map.go

```diff
@@ -314,3 +314,161 @@ func MergeJSON(a, b map[string]interface{}) (map[string]interface{}, error) {
 
 	return o, nil
 }
+
+// MergeStructs takes two input ValidatedGoStructs and merges their contents,
+// returning a new ValidatedGoStruct. If the input structs a and b are of
+// different types, an error is returned.
```

Comment:
It may be worth spelling out how the more complex situations involving slices pan out e.g. if structs ptr fields are equal but descendant list members differ.